android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt

```diff
@@ -711,34 +711,41 @@ constructor(
       }
 
       val todayCases = patients.filter {
-
-        if(it.extension?.get(0)?.value?.asStringValue()?.isNotEmpty() == true){
-          val date = convertToDateStringToDate(it.extension?.get(0)?.value?.asStringValue() ?: "")
-          date.isToday()
+        val extension = it?.extension?.find { it.url?.substringAfterLast("/").equals("patient-registraion-date") }
+        if(extension != null && extension.value?.asStringValue()?.isNotEmpty() == true){
```

Comment:
what is this if statement? can you detail what you're trying to do?

Reply:
Adding a check to make sure register date string is not empty otherwise date parser will throw exception.